contracts/refs/CoreRefV2.sol

```diff
@@ -31,21 +32,21 @@ abstract contract CoreRefV2 is ICoreRefV2, Pausable {
     /// 2. execute the code
     /// 3. call core and unlock the lock
     modifier globalLock(uint8 level) {
-        uint8 startingLevel = _core.lockLevel();
+        IGlobalReentrancyLock lock = globalReentrancyLock();
+        uint8 startingLevel = lock.lockLevel();
         require(
             startingLevel < level,
             "CoreRef: cannot lock less than current level"
         );
-        _core.lock(level);
+        lock.lock(level);
         _;
-        _core.unlock(startingLevel);
+        lock.unlock(startingLevel);
```

Comment:
We can simplify the logic here, since we refactored to only be able to go up 1 level : 
- Don't read lock.lockLevel(), don't require(...), this is enforced by the GlobalReentrancyLock already
- simplify function to `lock.lock(level); _; lock.unlock(level - 1);`